packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -930,20 +941,48 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
     }
 
     handleFilePreviewLoad = (response: InferDomainResponse, file: File) => {
-        const { domain, setFileImportData } = this.props;
+        const { domain, setFileImportData, domainFormDisplayOptions } = this.props;
+        const retainReservedFields = domainFormDisplayOptions?.retainReservedFields;
 
-        // if the DomainForm usage wants to show the file preview and import data options, then set these state values
-        if (response && response.fields.size > 0) {
+        let fields = List<QueryColumn>();
+        let reservedFields = response?.reservedFields || List<QueryColumn>();
+        if (retainReservedFields) {
+            fields = fields.merge(reservedFields);
+        }
+        if (response?.fields?.size) {
+            response.fields.forEach(field => {
+                if (domain.reservedFieldNames?.indexOf(field.name.toLowerCase()) < 0) {
```

Comment:
```suggestion
                if (!domain.isReservedFieldName(field.name)) {
```

With this as added to the models.DomainDesign:

    isReservedFieldName(fieldName: string): boolean {
        return (
            this.reservedFieldNames?.findIndex(
                rField => fieldName && rField.localeCompare(fieldName, 'en', { sensitivity: 'base' }) === 0
            ) >= 0
        );
    }

Reply:
Sorry, just seeing this now.  Though more robust and inclusive, it seems overkill for current purposes.  Since I'm on the verge or the merge, I'll leave it as is for now.